### ISSUE TITLE
Gather-drop failing to download dotnet-monitor

### ISSUE THREAD

**Issue by adiaaida** [dotnet/arcade#8442]
From gathering dotnet-monitor:

```
Gathering drop for build 20220204.6 of https://github.com/dotnet/dotnet-monitor
  Skipping non-shipping asset assets/manifests/dotnet-dotnet-monitor/7.0.0-preview.1.22104.6/MergedManifest.xml
  Downloading asset diagnostics/monitor7.0/release/dotnet-monitor.7.0.0-preview.1.22104.6.nupkg.version
    Failed to download asset, errors shown below:
      Blob uri 'https://dev.azure.com/dnceng/internal/_apis/build/builds/1593636/artifacts for diagnostics/monitor7.0/release/dotnet-monitor.7.0.0-preview.1.22104.6.nupkg.version is of an unknown type
      Blob uri 'https://dotnetbuilds.blob.core.windows.net/public for diagnostics/monitor7.0/release/dotnet-monitor.7.0.0-preview.1.22104.6.nupkg.version is of an unknown type
Aborting download.
  Downloading asset diagnostics/monitor7.0/release/dotnet-monitor.7.0.0-preview.1.22104.6.nupkg.buildversion
    Failed to download asset, errors shown below:
      Blob uri 'https://dev.azure.com/dnceng/internal/_apis/build/builds/1593636/artifacts for diagnostics/monitor7.0/release/dotnet-monitor.7.0.0-preview.1.22104.6.nupkg.buildversion is of an unknown type
      Blob uri 'https://dotnetbuilds.blob.core.windows.net/public for diagnostics/monitor7.0/release/dotnet-monitor.7.0.0-preview.1.22104.6.nupkg.buildversion is of an unknown type
Aborting download.
  Downloading asset diagnostics/monitor/7.0.0-preview.1.22104.6/dotnet-monitor.7.0.0-preview.1.22104.6.nupkg.sha512
    Failed to download asset, errors shown below:
      Blob uri 'https://dev.azure.com/dnceng/internal/_apis/build/builds/1593636/artifacts for diagnostics/monitor/7.0.0-preview.1.22104.6/dotnet-monitor.7.0.0-preview.1.22104.6.nupkg.sha512 is of an unknown type
      Blob uri 'https://dotnetbuilds.blob.core.windows.net/public-checksums for diagnostics/monitor/7.0.0-preview.1.22104.6/dotnet-monitor.7.0.0-preview.1.22104.6.nupkg.sha512 is of an unknown type
Aborting download.
  Downloading asset diagnostics/monitor/7.0.0-preview.1.22104.6/dotnet-monitor.7.0.0-preview.1.22104.6.nupkg
    Failed to download asset, errors shown below:
      Blob uri 'https://dev.azure.com/dnceng/internal/_apis/build/builds/1593636/artifacts for diagnostics/monitor/7.0.0-preview.1.22104.6/dotnet-monitor.7.0.0-preview.1.22104.6.nupkg is of an unknown type
      Blob uri 'https://dotnetbuilds.blob.core.windows.net/public for diagnostics/monitor/7.0.0-preview.1.22104.6/dotnet-monitor.7.0.0-preview.1.22104.6.nupkg is of an unknown type
Aborting download.
  Skipping non-shipping asset assets/symbols/Microsoft.Diagnostics.Monitoring.WebApi.7.0.0-preview.1.22104.6.symbols.nupkg
  Downloading asset assets/symbols/dotnet-monitor.7.0.0-preview.1.22104.6.symbols.nupkg
    Failed to download asset, errors shown below:
      Blob uri 'https://dev.azure.com/dnceng/internal/_apis/build/builds/1593636/artifacts for assets/symbols/dotnet-monitor.7.0.0-preview.1.22104.6.symbols.nupkg is of an unknown type
      Blob uri 'https://dotnetbuilds.blob.core.windows.net/public for assets/symbols/dotnet-monitor.7.0.0-preview.1.22104.6.symbols.nupkg is of an unknown type
Aborting download.
  Skipping non-shipping asset Microsoft.Diagnostics.Monitoring.WebApi@7.0.0-preview.1.22104.6
  Downloading asset dotnet-monitor@7.0.0-preview.1.22104.6
    https://pkgs.dev.azure.com/dnceng/public/_apis/packaging/feeds/dotnet-tools/nuget/packages/dotnet-monitor/versions/7.0.0-preview.1.22104.6/content =>
      D:\a\1\a\ReleaseTarget\dotnet-monitor\20220204.6\shipping\packages\dotnet-monitor.7.0.0-preview.1.22104.6.nupkg
      D:\a\1\a\ReleaseTarget\shipping\packages\dotnet-monitor.7.0.0-preview.1.22104.6.nupkg
Failed to download build with id 125772
```

Build: https://dev.azure.com/dnceng/internal/_build/results?buildId=1593636&view=logs&j=0405beeb-c9f5-5d9d-baf9-b8ef4da354d1&t=125c67c1-7a6f-516c-f4f0-1d91ec1d119d

**Comment by adiaaida** [comment]
I poked around the build, and it looks like it is using an old version of darc. Upgrading to latest should fix the problem as I'm pretty sure @alexperovich and I fixed this issue in https://github.com/dotnet/arcade-services/commit/5c1bc4f8a993e3f9f4d408f475d137cb179e21b0#diff-66d7e4aa8669364fa4e42010f203f4fa6ddc132b6d237db44b6ed8e2bd813d3b

**Comment by adiaaida** [comment]
The problem here was two fold:

1. dotnet-monitor had pinned the darc version, missing a lot of updates for the new location
2. A while back, we introduced the --separated flag to gather-drop, and changed the default behavior so that we did not pull down the files into repo-separated directories. This is a functionality that dotnet-monitor used, so they needed to add the --separated flag to their ci yml.
